validator/src/lib.rs

```diff
@@ -1,30 +1,33 @@
 extern crate valico;
 extern crate serde_json;
-
-use std::error::Error;
+#[macro_use] extern crate error_chain;
 
 use serde_json::{Value};
 use valico::json_schema;
 
-pub fn get_schema(api_version: &str) -> Result<Value, Box<Error>> {
+pub mod errors;
+
+use errors::*;
+
+pub fn get_schema(api_version: &str) -> Result<Value> {
     let schema = match api_version {
         "0.13" => include_str!("../schema/13.json"),
         "0.12" => include_str!("../schema/12.json"),
         "0.11" => include_str!("../schema/11.json"),
         "0.9" => include_str!("../schema/9.json"),
         "0.8" => include_str!("../schema/8.json"),
-        _ => return Err(format!("Invalid api version: {}", api_version).into())
-    }.parse()?;
+        _ => return Err(ErrorKind::WrongApiVersion(api_version.into()).into())
+    }.parse().expect(&format!("Parsing of schema for {} failed", api_version));
     Ok(schema)
 }
 
-pub fn validate_spaceapi_json(json: &str) -> Result<json_schema::ValidationState, Box<Error>> {
+pub fn validate_spaceapi_json(json: &str) -> Result<json_schema::ValidationState> {
```

Comment:
Alright. But it's confusing because all of a sudden `Result` only takes a single type parameter even though no `Result` was visibly imported anywhere.